libminifi/include/utils/TestUtils.h

```diff
@@ -0,0 +1,80 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+
+#include "../../test/TestBase.h"
+
+#ifdef WIN32
+#include <direct.h>
+#define GetCurrentDir _getcwd
+#else
+#include <unistd.h>
+#define GetCurrentDir getcwd
+#endif
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+
+std::string getCurrentWorkingDir(void) {
+  char buff[FILENAME_MAX];
+  GetCurrentDir(buff, FILENAME_MAX);
+  std::string current_working_dir(buff);
+  return current_working_dir;
+}
+
+std::string createTempDir(TestController* testController) {
+  char dirtemplate[] = "/tmp/gt.XXXXXX";
+  std::string temp_dir = testController->createTempDirectory(dirtemplate);
+  REQUIRE(!temp_dir.empty());
+  struct stat buffer;
+  REQUIRE(-1 != stat(temp_dir.c_str(), &buffer));
+  REQUIRE(S_ISDIR(buffer.st_mode));
+  return temp_dir;
+}
+
+std::string putFileToDir(const std::string& dir_path, const std::string& file_name, const std::string& content) {
+  std::string file_path(dir_path + utils::file::FileUtils::get_separator() + file_name);
+  std::ofstream out_file(file_path);
+  if (out_file.is_open()) {
+    out_file << content;
+  }
+  return file_path;
+}
+
+std::string createTempDirWithFile(TestController* testController, const std::string& file_name, const std::string& content) {
+  std::string temp_dir = createTempDir(testController);
+  putFileToDir(temp_dir, file_name, content);
+  return temp_dir;
```

Comment:
I would return the full path (ie. temp_dir + separator + file_name), as I expect that is what most users of this function will need.  I understand you can get the full path from temp_dir more easily than the other way round, but if a user needs the name of the directory, they can use `createTempDir` + `putFileToDir` instead.

Reply:
I think this was exactly what I needed: the name of directory with a single file in it, in order to have the processor `GetFile` exactly one target to run on. Maybe I should rename the function to indicate that this is giving back the path of a temp dir with a file?

Reply:
No, I think one version is enough; in either case, one can call `createTempDir` + `putFileToDir` and use the return value of either of these.

My reason for asking was that in my TailFile tests, I have a similar local utility function (and in those tests, I need the full path), and I could replace my local function with `createTempDirWithFile` if it returned the full path, too.

I think we don't have enough data to decide which is the more common use case, so I'm fine with leaving it as is; after some time, if we find that the TailFileTests kind of usage is the more common one, we can change it.